### DISCUSSION TITLE
ImmediateRenderer on X11/Linux

### DISCUSSION THREAD

**Discussion by adirh3** [AvaloniaUI/Avalonia#9347] [Q&A] [answered]
Hey,

Just wondering why it's not possible to use ImmediateRenderer on Linux/X11? (No UseDeferredRendering option for X11PlatformOptions)

Thanks!

**Comment by kekekeks** [comment]
It's considered to be a placeholder renderer for platforms that don't support the deferred one yet, since it has several issues, mostly with hit-testing. With X11 platform being done from scratch, it was initially implemented with DeferredRenderer, so there was no need for the immediate one.

**Comment by adirh3** [comment, marked as answer]
> It's considered to be a placeholder renderer for platforms that don't support the deferred one yet, since it has several issues, mostly with hit-testing. With X11 platform being done from scratch, it was initially implemented with DeferredRenderer, so there was no need for the immediate one.

I understand the situation but still I don't quite get why there was no need for the ImmediateRenderer as well, since as far as I understood it, it should perform better than the DeferredRenderer but consume more resources.
So why not let people choose as an option like Windows? 
Thanks

**Comment by timunie** [comment]
Converted this into Q&A as this is neither a bug nor a feature request. If you disagree please let us know.
